smalltalksrc/Slang/TMethod.class.st

```diff
@@ -2497,7 +2497,9 @@ TMethod >> recordDeclarationsIn: aCCodeGen [
 				declarationAt: varName
 				put: varType , varName ].
 		pragma selector = #returnTypeC: ifTrue: [ 
-			self returnType: pragma arguments last ] ]
+			self returnType: pragma arguments last ] ].
+	
+		cachedLocals := nil
```

Comment:
???

Reply:
I don't remember exactly where I stumbled onto this but the `allLocals` method consults the `cachedLocals`, if call that method right after recording declarations it will answer incorrectly, to avoid that I'm restart the cache here

Reply:
I don't understand why recording declarations break the locals cache. What is in that cache?

Anyway, we can add a comment :)

Reply:
The cache contains the combination of locals and declarations, the name doesn't help 😅 

```st
TMethod >> allLocals
	"The merge of locals and declarations (don't ask)"

	| allLocals |
	cachedLocals ifNotNil: [ ^ cachedLocals ].
	
	allLocals := Set new.
	allLocals addAll: (declarations keys asSet reject: [:k| (declarations at: k) == #implicit]).
	allLocals addAll: self locals.
	self nodesDo: [ :node |
		node isStatementList ifTrue: [ allLocals addAll: node locals ] ].
	^ cachedLocals := allLocals
```

Added a comment